From 1ea81a335baa08746df7daf2707c070271990937 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Pavel=20B=C5=99ezina?= <pbrezina@redhat.com>
Date: Wed, 22 Mar 2017 12:32:31 +0100
Subject: [PATCH 82/90] build: make curl required by secrets
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Also remove --disable-libcurl since it doesn't make sense.

Reviewed-by: Lukáš Slebodník <lslebodn@redhat.com>
Reviewed-by: Simo Sorce <simo@redhat.com>
Reviewed-by: Jakub Hrozek <jhrozek@redhat.com>
(cherry picked from commit 793f2573b2beaf8b48eab850429482acf68ec2b1)
---
 configure.ac            |  6 +++++-
 src/external/libcurl.m4 | 16 ++--------------
 2 files changed, 7 insertions(+), 15 deletions(-)

diff --git a/configure.ac b/configure.ac
index cf5e2557ef0a1bd6374200aa33abea6c509d03aa..80d8ea9ff5785b0d76edbb04f454d0dd8c8a1e6d 100644
--- a/configure.ac
+++ b/configure.ac
@@ -201,9 +201,13 @@ if test x$with_secrets = xyes; then
 fi
 
 if test x$with_kcm = xyes; then
-    m4_include([src/external/libcurl.m4])
     m4_include([src/external/libuuid.m4])
 fi
+
+if test x$with_kcm = xyes -o x$with_secrets = xyes; then
+    m4_include([src/external/libcurl.m4])
+fi
+
 # This variable is defined by external/libcurl.m4, but conditionals
 # must be always evaluated
 AM_CONDITIONAL([BUILD_WITH_LIBCURL],
diff --git a/src/external/libcurl.m4 b/src/external/libcurl.m4
index b420b04ad806bd1251f086b773ffe480d39f8bd3..42be308cd1e4b04e736daf887be9b75ea92db80e 100644
--- a/src/external/libcurl.m4
+++ b/src/external/libcurl.m4
@@ -1,17 +1,5 @@
-AC_ARG_ENABLE([curl],
-              [AS_HELP_STRING([--disable-curl-support],
-                              [do not build with libcurl support])],
-              [enable_libcurl=$enableval],
-              [enable_libcurl=yes])
-
-found_libcurl="no"
-AS_IF([test x$enable_libcurl = xyes],
-      [PKG_CHECK_MODULES([CURL],
-                         [libcurl],
-                         [found_libcurl=yes],
-                         [AC_MSG_ERROR([
-The libcurl development library was not found.])
-      ])])
+PKG_CHECK_MODULES([CURL], [libcurl], [found_libcurl=yes],
+              [AC_MSG_ERROR([The libcurl development library was not found.])])
 
 AS_IF([test x"$found_libcurl" = xyes],
     CFLAGS="$CFLAGS $CURL_CFLAGS"
-- 
2.9.3

